### PR TITLE
feat: consolidate tests

### PR DESCRIPTION
**Pull request by arayikhalatyan**
Resolves INT-4701

Note, had to `cargo update -p slab --precise 0.4.11` because it was failing `cargo audit`
I went through all the extension unit tests but not system and primitives yet
Added different `num_limb` tests to `algebra, ecc`
added edge case length tests to `keccak` and `sha2`

### CODE REVIEW

File: extensions/ecc/circuit/src/weierstrass_chip/tests.rs [hunk 1]
```diff
@@ -80,298 +86,642 @@ lazy_static::lazy_static! {
...
+            let y1 = y1 % modulus;
+            let x2 = x2 % modulus;
+            let y2 = y2 % modulus;
+            if rng.gen_bool(0.5) {
+                (x1, y1, x2, y2, Rv32WeierstrassOpcode::EC_ADD_NE as usize)
+            } else {
+                (x2, y2, x1, y1, Rv32WeierstrassOpcode::EC_ADD_NE as usize)
+            }
+        } else {
+            panic!("Random inputs are not valid for ec_addne");
```
[the remark this answers is not in the document]
Comment (jonathanpwang) [reply extensions/ecc/circuit/src/weierstrass_chip/tests.rs:182]: We need to assume that the input points are actually on the curve, which is why it's harder to generate random inputs generically. You can explain that in the panic message.

**Review by jonathanpwang — APPROVED** [review]
LGTM except small comments
